clib/memprof_coq.std.ml

```diff
@@ -0,0 +1,144 @@
+let is_interrupted _ = false [@@inline]
+
+module Resource_bind = struct
+  let ( let& ) f scope = f ~scope
+end
+
+module Mutex_aux = struct
+  external unlock: Mutex.t -> unit = "caml_mutex_unlock"
+
+  (* Critical sections :
+     - Mutex.lock does not poll on leaving the blocking section
+       since 4.12.
+     - Never inline, to avoid theoretically-possible reorderings with
+       flambda.
+     - Inline the call to Mutex.unlock to avoid polling in bytecode.
+       (workaround to the lack of masking) *)
+  let[@inline never] with_lock m ~scope =
+    let () = Mutex.lock m (* BEGIN ATOMIC *) in
+    match (* END ATOMIC *) scope () with
+    | (* BEGIN ATOMIC *) x -> unlock m ; (* END ATOMIC *) x
+    | (* BEGIN ATOMIC *) exception e -> unlock m ; (* END ATOMIC *) raise e
+
+end
+
+module Thread_map_core = struct
+  open Resource_bind
+
+  module IMap = Map.Make (
```

Comment:
is there a reason we can't use Int.Map here?

Reply:
Again code from `memprof-limits`, as this is just a temporal workaround I would stay close to the code, I can imagine we dropping this code once OCaml 5.3 is released.